paasta_tools/flink_tools.py

```diff
@@ -113,22 +112,11 @@ def load_flink_instance_config(
     general_config = service_configuration_lib.read_service_configuration(
         service, soa_dir=soa_dir
     )
-    flink_conf_file = "flink-%s" % cluster
-    instance_configs = service_configuration_lib.read_extra_service_information(
-        service, flink_conf_file, soa_dir=soa_dir
+    instance_config = load_service_instance_config(
+        service, instance, "flink", cluster, soa_dir=soa_dir
     )
-
-    if instance.startswith("_"):
-        raise InvalidJobNameError(
-            f"Unable to load kubernetes job config for {service}.{instance} as instance name starts with '_'"
-        )
-    if instance not in instance_configs:
-        raise NoConfigurationForServiceError(
-            f"{instance} not found in config file {soa_dir}/{service}/{flink_conf_file}.yaml."
-        )
-
     general_config = deep_merge_dictionaries(
-        overrides=instance_configs[instance], defaults=general_config
+        overrides=instance_config, defaults=general_config
     )
```

Comment:
I'm seeing a lot of copies of these three lines -- does it make sense to move them into `load_service_instance_config` so they can be DRYed?

Reply:
Yeah, I tried this, and the main reason I didn't was that the PaastaServiceConfigLoader needed something different. When I look into removing it, I'll consider this too.